lib/SIL/Utils/BasicBlockUtils.cpp

```diff
@@ -390,7 +390,15 @@ void DeadEndBlocks::compute() {
 //                  Post Dominance Set Completion Utilities
 //===----------------------------------------------------------------------===//
 
-void JointPostDominanceSetComputer::findJointPostDominatingSet(
+static bool endsInUnreachable(SILBasicBlock *block) {
```

Comment:
Can you refactor out the immediate logic related to this in DeadEndBlocks (maybe into a helper static function?) and use that here so we have one implementation.

Reply:
Also why not just use dead end blocks here? I want @atrick to also look at this.

Reply:
ok, I moved it to a static utility function in DeadEndBlocks.

> Also why not just use dead end blocks here?

Because it causes the problems I try to fix. E.g. in case the original dominatingBlock is already a dead-end block.